R/helpers.R

```diff
@@ -62,19 +62,19 @@ use_dependency <- function(package, type, min_version = NULL) {
       "!" = "Package {.pkg {package}} is already listed in
              {.field {existing_type}} in DESCRIPTION; no change made."
     ))
-  } else if (delta == 0 && !is.null(min_version)) {
+  } else if (delta == 0 && version_spec(version) != version_spec(existing_version)) {
     # change version
-    upgrade <- existing_version == "*" ||
-      numeric_version(min_version) > version_spec(existing_version)
-    if (upgrade) {
-      ui_bullets(c(
-        "v" = "Increasing {.pkg {package}} version to {.val {version}} in
-               DESCRIPTION."
-      ))
-      desc$set_dep(package, type, version = version)
-      desc$write()
-      changed <- TRUE
-    }
+    direction <- ifelse(version_spec(version) > version_spec(existing_version),
+                        "Increasing", "Decreasing")
+
+    ui_bullets(c(
+      "v" = "{.drn {direction}} {.pkg {package}} version to {.val {version}} in
```

Comment:
I'm not familiar with the `.drn` inline markup. Can you help me understand that?

Reply:
I just tried to replicate the same logic as in `{.pkg {package}}` and`{.val {version}}`. I tried to remove `.drn` and then `{direction}` would not resolve with the string in the object `direction`. Any hint?

Reply:
Something like `{.pkg {package}}` does interpolation of `package` through its surrounding brackets. Then `.pkg` and the second layer of brackets applies some inline styling.

For pure interpolation (which is the goal with `direction`), we just need one pair of brackets.

You can learn more about this in the cli package: https://cli.r-lib.org/reference/inline-markup.html.

Reply:
Thanks for pointing this out, and thanks again for your time and patience today!